site/docs/manual/Architecture.md

```diff
@@ -50,34 +50,77 @@ world = new World();
 
 ## Components
 A `Component` ([API Reference](/api/classes/component)) is an object that can store data but should have not behaviour (As that should be handled by systems). There is not a mandatory way to define a component.
-It could just be a function:
+
+To create a component you must extend the `Component` class and define a schema:
 ```javascript
-function ComponentA() {
-  this.number = 10;
-  this.string = "Hello";
+import { Component, Types } from 'ecsy';
+
+class ComponentA extends Component {}
+
+ComponentA.schema = {
+  number: { type: Types.Number, default: 10 },
+  string: { type: Types.String, default: "Hello" }
 }
 ```
 
-The recommended way is using ES6 classes and extending the `Component` class:
+The schema is used to set the default values of a component. ECSY also uses it to implement the default `.copy()`, `.clone()`, and `.reset()` methods. Schemas can also be used for tooling and serialization, something we plan on covering in the future.
```

Comment:
maybe mentioning that is also using for pooling? just so people understand why this is currently important outside of tooling?
Also do you think is it worth to mention that you it's totally ok if you want to implement your own functions?

Reply:
Yup, and I cover the fact that you can implement your own types later on.